### ISSUE TITLE
what does iteration mean in the configuration file? 

### ISSUE THREAD

**Issue by 12neurotool** [NifTK/NiftyNet#438]
I am not clear about what the iteration means. Could you explain more about it? Also, how is it related to epoch?

Thanks.

**Comment by fepegar** [comment]
At each iteration, [`batch_size` windows are extracted from the queue](https://niftynet.readthedocs.io/en/dev/config_spec.html#queue-length), they're fed to the network and the weights are updated using the gradient of the loss function with respect to the weights.

The concept of epoch doesn't apply in this case, because NiftyNet uses [patch-based training](https://niftynet.readthedocs.io/en/dev/window_sizes.html) (patch and window are the same thing).
